### PR TITLE
SW-616_do_not_override_components_if_fl_is_inactive

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -70,4 +70,4 @@ jobs:
       - name: Run test suite
         run: |
           cd development/custom/plugins/plugin-shopware-6
-          composer test
+          SYMFONY_DEPRECATIONS_HELPER=disabled composer test
```
Comment (TheKeymaster) [inline .github/workflows/phpunit.yml:73]: Shopware internally has some deprecations, which are way too many that we can get rid of, so we need to disable the deprecation helper again, unfortunately.

File: src/Resources/views/storefront/component/listing/filter/filter-multi-select.html.twig [hunk 2]
```diff
@@ -1,16 +1,20 @@
 {% sw_extends '@Storefront/storefront/component/listing/filter/filter-multi-select.html.twig' %}
 
 {% block component_filter_multi_select %}
-    <div class="{% if hidden %}d-none {% endif %}filter-multi-select filter-multi-select-{{ name }} filter-panel-item{% if not sidebar %} dropdown{% endif %}"
-         data-{{ pluginSelector }}="true"
-         data-{{ pluginSelector }}-options='{{ dataPluginSelectorOptions|json_encode }}'>
+    {% if not context.context.extensions.findologicService.enabled %}
```
Comment (dblaichinger) [inline src/Resources/views/storefront/component/listing/filter/filter-multi-select.html.twig:4]: Does enabled stand for navigation?

File: tests/Struct/FindologicProductTest.php [hunk 3]
```diff
@@ -369,9 +368,13 @@ public function testProduct(): void
         );
 
         $salesChannel = $this->salesChannelContext->getSalesChannel();
-        $domain = $salesChannel->getDomains()->first()->getUrl();
+        $domains = $salesChannel->getDomains();
```
Comment (dblaichinger) [inline tests/Struct/FindologicProductTest.php:371]: Is there already a `default.headless` included?

**Review by dblaichinger — COMMENTED** [review]
Two questions

File: src/Resources/views/storefront/component/listing/filter/filter-multi-select.html.twig (hunk 2 above)
Comment (TheKeymaster) [reply src/Resources/views/storefront/component/listing/filter/filter-multi-select.html.twig:4]: The `FindologicService` is aware of the state and tells us if Findologic should be used or not.

File: tests/Struct/FindologicProductTest.php (hunk 3 above)
Comment (TheKeymaster) [reply tests/Struct/FindologicProductTest.php:371]: Yes that's the Headless Sales Channel, which is always created by default. Only recently with Shopware 6.4 it appears to also to include an URL which is not really an URL, so we just filter all of these headless Sales Channels out of this.

**Pull request merged**